index.js

```diff
@@ -67,10 +64,10 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
         if (ev.total) currentSize = ev.total
       })
 
-      upload.send(function (err) {
+      upload.send(function (err, data) {
         if (err) return cb(err)
 
-        cb(null, { size: currentSize, key: filePath })
+        cb(null, extend(data, { size: currentSize, key: key }))
```

Comment:
Also, `data` has keys with a capital letter which is not consistent with any javascript style guide I have ever seen. I might have been wrong about the amount of information though, how about the following?

```javascript
cb(null, {
  size: currentSize,
  key: key,
  location: data.Location,
  etag: data.ETag
})
```